app/src/main/java/de/tum/in/tumcampusapp/widgets/MVVWidget.java

```diff
@@ -0,0 +1,146 @@
+        for (int appWidgetId : appWidgetIds) {
+            updateAppWidget(context, appWidgetManager, appWidgetId);
+        }
+        setAlarm(context);
+        super.onUpdate(context, appWidgetManager, appWidgetIds);
+    }
+
+    @Override
+    public void onDeleted(Context context, int[] appWidgetIds) {
+        // When the user deletes the widget, delete the associated setting from the database.
+        TransportManager transportManager = new TransportManager(context);
+        for (int appWidgetId : appWidgetIds) {
+            transportManager.deleteWidget(appWidgetId);
+        }
+        super.onDeleted(context, appWidgetIds);
+    }
+
+    @Override
+    public void onEnabled(Context context) {
+        // Enter relevant functionality for when the first widget is created
+        setAlarm(context);
+    }
+
+    @Override
+    public void onDisabled(Context context) {
+        // Cancel alarm as the last widget has been removed
+        Intent intent = new Intent(context, MVVWidget.class);
+        PendingIntent sender = PendingIntent.getBroadcast(context, 0, intent, 0);
+        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        alarmManager.cancel(sender);
+        alarmIsSet = false;
+        super.onDisabled(context);
+    }
+
+    /**
+     * If no alarm is running yet a new repeating alarm is started
+     */
+    private static void setAlarm(Context context) {
+        if (alarmIsSet) return;
+        AlarmManager am = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        Intent intent = new Intent(context, MVVWidget.class);
+        intent.setAction(BROADCAST_ALARM_NAME);
+        PendingIntent pi = PendingIntent.getBroadcast(context, 0, intent, 0);
+        //After after 30 or 60 (android may delay) seconds
+        am.setRepeating(AlarmManager.RTC, System.currentTimeMillis(), 30000, pi);
+        alarmIsSet = true;
+    }
+
+    /**
+     * Updates the content of the widget
+     *
+     * @param appWidgetId the id of the widget to update
+     */
+    static void updateAppWidget(Context context, AppWidgetManager appWidgetManager,
+                                int appWidgetId) {
+        String station = context.getString(R.string.mvv_widget_no_station);
+        String station_id = "";
+
+        // Get the settings for this widget from the database
+        TransportManager transportManager = new TransportManager(context);
+        Cursor c = transportManager.getWidget(appWidgetId);
+        if (c.getCount() >= 1) {
+            c.moveToFirst();
+            station = c.getString(c.getColumnIndex("station"));
+            station_id = c.getString(c.getColumnIndex("station_id"));
+            Boolean use_location = c.getInt(c.getColumnIndex("location")) != 0;
+            c.close();
+            if (use_location) {
+                // TODO implement nearest station (replace the station_id string with the calculated station)
```

Comment:
What's the plan for this feature?

As I see this, we would need to add another endpoint in the TransportManager, i.e. a new `XML_STOPFINDER_REQUEST` with the current coordinates:

Maybe similar to this?:
http://efa.vrr.de/vrr/XML_STOPFINDER_REQUEST?language=de&type_sf=coord&name_sf=3435366.4594533667:463680.7198482931:NAV3:5